vowpalwabbit/interactions_predict.h

```diff
@@ -91,8 +91,8 @@ namespace INTERACTIONS
   // this templated function generates new features for given example and set of interactions
   // and passes each of them to given function T()
   // it must be in header file to avoid compilation problems
-  template <class R, class S, void(*T)(R&, float, S), bool audit, void(*audit_func)(R&, const audit_strings*), class W> // nullptr func can't be used as template param in old compilers
-  inline void generate_interactions(v_array<v_string>& interactions, bool permutations, example_predict& ec, R& dat, W& weights) // default value removed to eliminate ambiguity in old complers
+  template <class R, class S, void(*T)(R&, float, S), bool audit, void(*audit_func)(R&, const audit_strings*), class W, class I> // nullptr func can't be used as template param in old compilers
+  inline void generate_interactions(I& interactions, bool permutations, example_predict& ec, R& dat, W& weights) // default value removed to eliminate ambiguity in old complers
```

Comment:
Why do we need a class I?  It seems to be always a v_array<v_string>?

Reply:
not for the slim predict. I'm passing vectors to get allocation/deallocation safety in case of exceptions

Reply:
So, make it vector everyhwere?

Reply:
done

Reply:
done